src/modules/compliance/src/lib/procedures/EnsureInteractiveUsersHomeDirectoriesAreConfigured.cpp

```diff
@@ -0,0 +1,193 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+#include "IterateUsers.h"
+
+#include <CommonUtils.h>
+#include <Evaluator.h>
+#include <Result.h>
+#include <fcntl.h>
+#include <fstream>
+#include <grp.h>
+#include <pwd.h>
+#include <set>
+#include <string>
+#include <sys/stat.h>
+#include <unistd.h>
+
+namespace compliance
+{
+using std::ifstream;
+using std::map;
+using std::ostringstream;
+using std::set;
+using std::string;
+
+Result<Status> AuditEnsureFilePermissions(map<string, string>, IndicatorsTree&, ContextInterface&);
+Result<Status> RemediateEnsureFilePermissions(map<string, string>, IndicatorsTree&, ContextInterface&);
+
+namespace
+{
+constexpr const char* etcShellsPath = "/etc/shells";
+Result<set<string>> ListValidShells(ContextInterface& context)
+{
+    set<string> validShells;
+    ifstream shellsFile("/etc/shells");
+    if (!shellsFile.is_open())
+    {
+        OsConfigLogError(context.GetLogHandle(), "Failed to open %s file", etcShellsPath);
+        return Error(std::string("Failed to open ") + etcShellsPath + " file", EINVAL);
+    }
+    string line;
+    while (std::getline(shellsFile, line))
```

Comment:
This line can be simplified without an additional `using` due to argument dependent lookup:
```suggestion
    while (getline(shellsFile, line))
```